libraries/rhel_handler.rb

```diff
@@ -61,32 +53,72 @@ def configure_package(action_name)
           include_recipe "yum-chef::#{new_resource.channel}"
         end
 
-        # Foodcritic doesn't like timeout attribute in package resource
-        package new_resource.product_name do # ~FC009
-          action action_name
-          package_name ingredient_package_name
-          if use_custom_repo_recipe?
-            # Respect the options that the user has specified
-            options new_resource.options
-          else
-            # Ensure that we are installing from the correct repository
-            options "--disablerepo=* --enablerepo=chef-#{new_resource.channel} #{new_resource.options}"
-          end
-
-          # If the latest version is specified, we should not give any version
-          # to the package resource.
-          unless version_latest?(new_resource.version)
-            version version_for_package_resource
-          end
-
-          timeout new_resource.timeout
-
-          if new_resource.product_name == 'chef'
-            # We define this resource in ChefIngredientProvider
-            notifies :run, 'ruby_block[stop chef run]', :immediately
-          end
+        configure_from_repo(action_name)
+      end
+    end
+
+    def configure_from_source_package(action_name, local_path = nil)
+      rpm_package new_resource.product_name do
+        action action_name
+        package_name ingredient_package_name
+        options new_resource.options
+        source local_path || new_resource.package_source
+
+        if new_resource.product_name == 'chef'
+          # We define this resource in ChefIngredientProvider
+          notifies :run, 'ruby_block[stop chef run]', :immediately
+        end
+      end
+    end
+
+    def configure_from_repo(action_name)
+      # Foodcritic doesn't like timeout attribute in package resource
+      package new_resource.product_name do # ~FC009
+        action action_name
+        package_name ingredient_package_name
+        if use_custom_repo_recipe?
+          # Respect the options that the user has specified
+          options new_resource.options
+        else
+          # Ensure that we are installing from the correct repository
+          options "--disablerepo=* --enablerepo=chef-#{new_resource.channel} #{new_resource.options}"
+        end
+
+        # If the latest version is specified, we should not give any version
+        # to the package resource.
+        unless version_latest?(new_resource.version)
+          version version_for_package_resource
+        end
+
+        timeout new_resource.timeout
+
+        if new_resource.product_name == 'chef'
+          # We define this resource in ChefIngredientProvider
+          notifies :run, 'ruby_block[stop chef run]', :immediately
         end
       end
     end
+
+    def configure_from_unstable_channel(action_name)
```

Comment:
We can maybe create a `HandlerBase` class and move this method over there. I do not think it is necessary since we only have one method but if you think it would be helpful we can :smile: